src/instrument/utils/exceptions.py

```diff
@@ -0,0 +1,133 @@
+"""Exception specific files for Bluesky engine"""
+
+from ophydregistry.exceptions import (  # noqa: F401
+    ComponentNotFound,
+    InvalidComponentLabel,
+    MultipleComponentsFound,
+)
+
+
+class NoEnergies(ValueError):
+    """The plan parameters do not result in any energies being selected for scanning."""
+
+    ...
+
+
+class GainOverflow(RuntimeError):
+    """The gain is trying to be set to a value that is not allowed."""
+
+    ...
+
+
+class FileNotWritable(IOError):
+    """Output file is available but does not have write intent."""
+
+    ...
+
+
+class XDIFilenameKeyNotFound(KeyError):
+    """The format string for an XDI filename has unknown placeholders."""
+
+    ...
+
+
+class InvalidPV(ValueError):
+    """A process variable path or similar is expected, but the path
+    provided is not valid.
+
+    """
+
+    ...
+
+
+class DocumentNotFound(RuntimeError):
+    """An attempt was made to use a document, but the requested document
+    was not available.
+
+    """
+
+    ...
+
+
+class IOCTimeout(RuntimeError):
```

Comment:
Bluesky does not start any IOCs.  Why would XPCS's bluesky sessions need this exception?